File: organized-skills/1password-skill/tools-python/src/op_env/op_env_list.py
"""op-env-list — List environment items from 1Password (Python SDK version).

Usage:
    op-env-list [--vault <vault>] [--tags <tags>] [--json]
"""

from __future__ import annotations

import argparse
import asyncio
import json

from .utils import colors, get_client, list_environments

HELP_EPILOG = """\
examples:
  # List all environments
  uv run op-env-list

  # Filter by vault
  uv run op-env-list --vault Development

  # Filter by tags
  uv run op-env-list --tags environment,production

  # JSON output
  uv run op-env-list --json
"""


def build_parser() -> argparse.ArgumentParser:
    parser = argparse.ArgumentParser(
        prog="op-env-list",
        description="List environment items from 1Password",
        epilog=HELP_EPILOG,
        formatter_class=argparse.RawDescriptionHelpFormatter,
    )
    parser.add_argument("--vault", help="Filter by vault name")
    parser.add_argument(
        "--tags", default="environment", help="Filter by tags (default: environment)"
    )
    parser.add_argument("--json", dest="json_output", action="store_true", help="JSON output")
    return parser


async def main() -> None:
    parser = build_parser()
    args = parser.parse_args()

    # list_environments uses the op CLI (hybrid approach — SDK lacks tag filtering)
    # We still call get_client() to validate the token is set, but the listing
    # itself goes through the CLI.
    _ = await get_client()

    items = await list_environments(vault=args.vault, tags=args.tags)

    # JSON output mode
    if args.json_output:
        print(json.dumps(items, indent=2))
        return

    if not items:
        print("No environments found")
        print()
        print("Tip: Create one with: uv run op-env-create <name> <vault> KEY=value")
        return

    c = colors

    print()
    print(c.cyan("+" + "=" * 79 + "+"))
    print(f"{c.cyan('|')}{'1Password Environments':^79}{c.cyan('|')}")
    print(c.cyan("+" + "=" * 79 + "+"))

    # Header
    name_h = "NAME".ljust(35)
    vault_h = "VAULT".ljust(20)
    updated_h = "UPDATED".ljust(15)
    print(f"{c.cyan('|')} {name_h} | {vault_h} | {updated_h} {c.cyan('|')}")
    print(c.cyan("+" + "=" * 79 + "+"))

    # Items
    for item in items:
        title = (item.get("title", "") or "")[:35].ljust(35)
        vault_name = (item.get("vault", {}).get("name", "") or "")[:20].ljust(20)
        updated = (item.get("updated_at", "N/A") or "N/A")[:10].ljust(15)
        print(f"{c.cyan('|')} {title} | {vault_name} | {updated} {c.cyan('|')}")

    print(c.cyan("+" + "=" * 79 + "+"))
    print()
    print(f"Total: {len(items)} environments")
    print()
    print("Commands:")
    print("  Show details: uv run op-env-show <name> <vault>")
    print("  Export .env:  uv run op-env-export <name> <vault> > .env")


def main_sync() -> None:
    asyncio.run(main())


if __name__ == "__main__":
    main_sync()

File: organized-skills/1password-skill/tools/src/op-env-list.ts
#!/usr/bin/env bun
/**
 * op-env-list - List environment items from 1Password
 *
 * Usage:
 *   op-env-list [--vault <vault>] [--tags <tags>]
 */

import { parseArgs } from "util";
import {
  log,
  colors,
  ensureAuth,
  listEnvironments,
} from "./utils";

const HELP = `
op-env-list - List environment items from 1Password

USAGE:
    op-env-list [OPTIONS]

OPTIONS:
    --vault <vault>   Filter by vault name
    --tags <tags>     Filter by tags (comma-separated, default: environment)
    --json            Output in JSON format
    --help            Show this help message

EXAMPLES:
    # List all environments
    op-env-list

    # Filter by vault
    op-env-list --vault Development

    # Filter by tags
    op-env-list --tags environment,production

    # JSON output
    op-env-list --json
`;

async function main() {
  const { values } = parseArgs({
    args: Bun.argv.slice(2),
    options: {
      vault: { type: "string" },
      tags: { type: "string", default: "environment" },
      json: { type: "boolean" },
      help: { type: "boolean", short: "h" },
    },
    allowPositionals: true,
  });

  if (values.help) {
    console.log(HELP);
    process.exit(0);
  }

  await ensureAuth();

  // Get environments
  const items = await listEnvironments(values.vault, values.tags);

  // JSON output mode
  if (values.json) {
    console.log(JSON.stringify(items, null, 2));
    process.exit(0);
  }

  if (items.length === 0) {
    console.log("No environments found");
    console.log();
    console.log("Tip: Create one with: op-env-create <name> <vault> KEY=value");
    process.exit(0);
  }

  const c = colors.cyan;

  console.log();
  console.log(c("╔═══════════════════════════════════════════════════════════════════════════════╗"));
  console.log(`${c("║")}                           1Password Environments                              ${c("║")}`);
  console.log(c("╠═══════════════════════════════════════════════════════════════════════════════╣"));

  // Header
  const nameHeader = "NAME".padEnd(35);
  const vaultHeader = "VAULT".padEnd(20);
  const updatedHeader = "UPDATED".padEnd(15);
  console.log(`${c("║")} ${nameHeader} │ ${vaultHeader} │ ${updatedHeader} ${c("║")}`);
  console.log(c("╠═══════════════════════════════════════════════════════════════════════════════╣"));

  // Items
  for (const item of items) {
    const title = (item.title || "").substring(0, 35).padEnd(35);
    const vault = (item.vault?.name || "").substring(0, 20).padEnd(20);
    const updated = (item.updated_at || "N/A").substring(0, 10).padEnd(15);
    console.log(`${c("║")} ${title} │ ${vault} │ ${updated} ${c("║")}`);
  }

  console.log(c("╚═══════════════════════════════════════════════════════════════════════════════╝"));
  console.log();
  console.log(`Total: ${items.length} environments`);
  console.log();
  console.log("Commands:");
  console.log("  Show details: op-env-show <name> <vault>");
  console.log("  Export .env:  op-env-export <name> <vault> > .env");
}

main().catch((err) => {
  log.error(err.message);
  process.exit(1);
});
